tests/Controller/TaxonApiTest.php

```diff
@@ -0,0 +1,335 @@
+    ];
+
+    /**
+     * @var array
+     */
+    private static $authorizedHeaderWithAccept = [
+        'HTTP_Authorization' => 'Bearer SampleTokenNjZkNjY2MDEwMTAzMDkxMGE0OTlhYzU3NzYyMTE0ZGQ3ODcyMDAwM2EwMDZjNDI5NDlhMDdlMQ',
+        'ACCEPT' => 'application/json',
+    ];
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_taxon_list_when_access_is_denied()
+    {
+        $this->loadFixturesFromFile('resources/taxons.yml');
+        $this->client->request('GET', '/api/v1/taxons/');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_taxon_when_it_does_not_exist()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('GET', '/api/v1/taxons/-1', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_taxons()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/taxons.yml');
+
+
+        $this->client->request('GET', '/api/v1/taxons/', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_taxon()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $taxons = $this->loadFixturesFromFile('resources/taxons.yml');
+
+        $this->client->request('GET', '/api/v1/taxons/'.$taxons['Women']->getId(), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_delete_taxon_if_it_does_not_exist()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('DELETE', '/api/v1/taxons/-1', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_delete_taxon()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $taxons = $this->loadFixturesFromFile('resources/taxons.yml');
+
+        $this->client->request('DELETE', '/api/v1/taxons/'.$taxons['Men']->getId(), [], [], static::$authorizedHeaderWithContentType, []);
```

Comment:
The last array is not needed. Also we could extract `$taxons['Men']` to the variable. It will be cleaner + with docblock some IDE's can allow for autocorrect.

```php
/** @var TaxonInterface $taxon */
$taxon = $taxons['Men'];
```